test/unittest/tests/block/test_block_session_size.cpp

```diff
@@ -0,0 +1,277 @@
+ * created or discarded.
+ */
+#include "wt_internal.h"
+#include <catch2/catch.hpp>
+#include "../wrappers/mock_session.h"
+
+void
+validate_size_block(WT_SIZE *size)
+{
+    REQUIRE(size != nullptr);
+    REQUIRE(size->depth == 0);
+    REQUIRE(size->off[0] == nullptr);
+    REQUIRE(size->size == 0);
+}
+
+void
+free_size_block(WT_SIZE *size)
+{
+    __wt_free(nullptr, size);
+}
+
+void
+validate_and_free_size_block(WT_SIZE *size)
+{
+    validate_size_block(size);
+    free_size_block(size);
+}
+
+void
+free_size_list(WT_BLOCK_MGR_SESSION *bms)
+{
+    WT_SIZE *curr = bms->sz_cache;
+    while (curr != NULL) {
+        WT_SIZE *tmp = curr;
+        curr = curr->next[0];
+        __wt_free(nullptr, tmp);
+    }
+}
+void
+validate_size_list(WT_BLOCK_MGR_SESSION *bms, int expected_items)
+{
+    REQUIRE(bms != nullptr);
+    if (bms->sz_cache_cnt == 0)
+        REQUIRE(bms->sz_cache == nullptr);
+
+    REQUIRE(bms->sz_cache_cnt == expected_items);
+    WT_SIZE *curr = bms->sz_cache;
+    for (int i = 0; i < expected_items; i++) {
+        validate_size_block(curr);
+        curr = curr->next[0];
+    }
+    REQUIRE(curr == nullptr);
+}
+
+void
+validate_and_free_size_list(WT_BLOCK_MGR_SESSION *bms, int expected_items)
+{
+    validate_size_list(bms, expected_items);
+    free_size_list(bms);
+}
+
+TEST_CASE("Block session: __block_size_alloc", "[block_session_size]")
+{
+    WT_SIZE *sz = nullptr;
+
+    REQUIRE(__ut_block_size_alloc(nullptr, &sz) == 0);
+    validate_and_free_size_block(sz);
+}
+
+TEST_CASE("Block session: __block_size_prealloc", "[block_session_size]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
+
+    SECTION("Allocate zero size blocks")
+    {
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 0) == 0);
+        validate_and_free_size_list(bms, 0);
+    }
+
+    SECTION("Allocate one size block")
+    {
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 1) == 0);
+        validate_and_free_size_list(bms, 1);
+    }
+
+    SECTION("Allocate multiple size blocks")
+    {
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 3) == 0);
+        validate_and_free_size_list(bms, 3);
+    }
+
+    SECTION("Allocate blocks on existing cache")
+    {
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 3) == 0);
+        validate_size_list(bms, 3);
+
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 0) == 0);
+        validate_size_list(bms, 3);
+
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 2) == 0);
+        validate_size_list(bms, 3);
+
+        REQUIRE(__ut_block_size_prealloc(session->getWtSessionImpl(), 5) == 0);
+        validate_and_free_size_list(bms, 5);
+    }
+}
+
+TEST_CASE("Block session: __wti_block_size_alloc", "[block_session_size]")
+{
```

Comment:
The tests in this TEST_CASE all look very useful, but they are not obvious in every case why they do what they do and why the expected results are as they are. I think more comments explaining each test, what it's trying to show and what the expected results are would be very useful.

Reply:
added comments [694803d](https://github.com/wiredtiger/wiredtiger/pull/10982/commits/694803d72c116470535d7523a607ff49d04894da)

Reply:
Thanks!